compiler/src/dotty/tools/dotc/reporting/trace.scala

```diff
@@ -50,7 +62,7 @@ object trace:
                          printer: Printers.Printer = Printers.default,
                          showOp: Any => String = alwaysToString)
                         (op: => T)(using Context): T =
-    if ctx.mode.is(Mode.Printing) || (printer eq Printers.noPrinter) then op
+    if ctx.mode.is(Mode.Printing) || !doForceTrace && (printer eq Printers.noPrinter) then op
```

Comment:
I am against having an override for printers. `trace.force` with default arguments gives you a printer anyway. If you want another printer, just enable it in Printers.scala. One way to do things.

Reply:
I think that printers are orthogonal to what `trace.force` is trying to enable.

What I really want to be able to do is to tell students "to see information about GADT constraints, force [this trace](https://github.com/dotty-staging/dotty/blob/fix-trace-force/compiler/src/dotty/tools/dotc/core/PatternTypeConstrainer.scala#L208)". And "if you need even more information, force [this one](https://github.com/dotty-staging/dotty/blob/fix-trace-force/compiler/src/dotty/tools/dotc/core/TypeComparer.scala#L209) as well". If the students were required to also enable the corresponding printers, that wouldn't give them any useful information - in fact, it'd just drown them in irrelevant information from the typer printer.

So what I'm trying to illustrate with that example is that I want something which allows forcing a specific trace to be printed out without doing anything else, and that something is `trace.force`.

Reply:
OK, I see the workflow better now. So the idea is that to enable a specific trace, you just replace the `trace` with `trace.force`. Makes sense.